tests/foreman/ui_airgun/test_repository.py

```diff
@@ -482,3 +493,279 @@ def test_positive_resynchronize_puppet_repo(session, module_prod):
+    """Perform end to end testing for custom yum repository
+
+    :id: 8baf11c9-019e-4625-a549-ec4cd9312f75
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    repo_name = gen_string('alpha')
+    checksum_type = CHECKSUM_TYPE['sha256']
+    new_repo_name = gen_string('alphanumeric')
+    new_checksum_type = CHECKSUM_TYPE['sha1']
+    gpg_key = entities.GPGKey(
+        content=read_data_file(VALID_GPG_KEY_FILE),
+        organization=module_org
+    ).create()
+    new_gpg_key = entities.GPGKey(
+        content=read_data_file(VALID_GPG_KEY_BETA_FILE),
+        organization=module_org,
+    ).create()
+    with session:
+        session.repository.create(
+            module_prod.name,
+            {
+                'name': repo_name,
+                'repo_type': REPO_TYPE['yum'],
+                'repo_content.upstream_url': FAKE_1_YUM_REPO,
+                'repo_content.checksum_type': checksum_type,
+                'repo_content.gpg_key': gpg_key.name,
+                'repo_content.download_policy': DOWNLOAD_POLICIES['on_demand']
+            }
+        )
+        assert session.repository.search(module_prod.name, repo_name)[0]['Name'] == repo_name
+        repo_values = session.repository.read(module_prod.name, repo_name)
+        assert repo_values['repo_content']['upstream_url'] == FAKE_1_YUM_REPO
+        assert repo_values['repo_content']['metadata_type'] == checksum_type
+        assert repo_values['repo_content']['gpg_key'] == gpg_key.name
+        assert repo_values['repo_content']['download_policy'] == DOWNLOAD_POLICIES['on_demand']
+        session.repository.update(
+            module_prod.name,
+            repo_name,
+            {
+                'name': new_repo_name,
+                'repo_content.upstream_url': FAKE_2_YUM_REPO,
+                'repo_content.metadata_type': new_checksum_type,
+                'repo_content.gpg_key': new_gpg_key.name,
+                'repo_content.download_policy': DOWNLOAD_POLICIES['immediate'],
+            }
+        )
+        assert not session.repository.search(module_prod.name, repo_name)
+        repo_values = session.repository.read(module_prod.name, new_repo_name)
+        assert repo_values['name'] == new_repo_name
+        assert repo_values['repo_content']['upstream_url'] == FAKE_2_YUM_REPO
+        assert repo_values['repo_content']['metadata_type'] == new_checksum_type
+        assert repo_values['repo_content']['gpg_key'] == new_gpg_key.name
+        assert repo_values['repo_content']['download_policy'] == DOWNLOAD_POLICIES['immediate']
+        session.repository.delete(module_prod.name, new_repo_name)
+        values = session.repository.search(module_prod.name, new_repo_name)
+        # a task is created for delete operation and we are not redirected to that task page.
+        # this operation can take some time.
+        for _ in range(10):
```

Comment:
can we check task status?

In case we have no other options:
```
        wait_for(
            lambda: session.repository.search(module_prod.name, new_repo_name) is None
            timeout=20,
            delay=2,
        )
```

Reply:
prefer to implement it at airgun side instead